### PR TITLE
Tests to go with the recent watcher refactoring

### PR DESCRIPTION
**Pull request by yahgwai**

### CODE REVIEW

File: src/watcher/garbageCollector.ts [hunk 1]
```diff
@@ -23,52 +24,74 @@ export class AppointmentStoreGarbageCollector {
...
-    private collecting = false;
+    /**
+     * Stop monitoring for expired events
+     */
+    public stop() {
+        if (this.started) {
+            this.started = false;
+            this.provider.removeListener("block", this.listener);
+        }
+    }
```
Comment (bigspider) [inline src/watcher/garbageCollector.ts]: Should we throw an error if start is called when it's already started, or stop when not started? Isn't it likely a bug if that happens?

File: src/watcher/garbageCollector.ts [hunk 2]
```diff
@@ -23,52 +24,74 @@ export class AppointmentStoreGarbageCollector {
...
 
     /**
      * Find appointments that have expired then remove them from the subsciber and store.
      * @param blockNumber
```
Comment (bigspider) [inline src/watcher/garbageCollector.ts]: typo: subscriber

File: src/watcher/garbageCollector.ts (hunk 1 above)
Comment (yahgwai) [reply src/watcher/garbageCollector.ts]: cool, I'll update

File: src/watcher/garbageCollector.ts (hunk 2 above)
Comment (yahgwai) [reply src/watcher/garbageCollector.ts]: updated

File: src/watcher/garbageCollector.ts (hunk 1 above)
Comment (yahgwai) [reply src/watcher/garbageCollector.ts]: I'll throw error for "start" but not for "stop" as errors in destructors can be dangerous. But I'll log error there.

**Pull request merged**